source/databricks/calculation_engine/tests/features/conftest.py

```diff
@@ -64,3 +69,248 @@ def actual_and_expected_views(
     scenario_path = str(Path(request.module.__file__).parent)
     executor = ViewScenarioExecutor(spark)
     return executor.execute(scenario_path)
+
+@pytest.fixture(scope="module")
+def test_cases(spark: SparkSession, request: pytest.FixtureRequest) -> TestCases:
+    """Fixture used for scenario tests. Learn more in package `testcommon.etl`."""
+
+    # Get the path to the scenario
+    scenario_path = str(Path(request.module.__file__).parent)
+    calculation_args = create_calculation_args(f"{scenario_path}/when/")
+
+    # Read input data
+    time_series_points = read_csv(
+        spark,
+        f"{scenario_path}/when/time_series_points.csv",
+        time_series_points_schema,
+    )
+    time_series_points = spark.createDataFrame(time_series_points.rdd, schema=time_series_points_schema, verifySchema=True)
+
+    grid_loss_metering_points = read_csv(
+        spark,
+        f"{scenario_path}/when/grid_loss_metering_points.csv",
+        grid_loss_metering_point_ids_schema,
+    )
+    grid_loss_metering_points = spark.createDataFrame(grid_loss_metering_points.rdd, schema=grid_loss_metering_point_ids_schema, verifySchema=True)
+
+    metering_point_periods = read_csv(
+        spark,
+        f"{scenario_path}/when/metering_point_periods.csv",
+        metering_point_periods_schema,
+    )
+    metering_point_periods = spark.createDataFrame(metering_point_periods.rdd, schema=metering_point_periods_schema, verifySchema=True)
+
+    # Defining the mocks for the data frames in the "when" folder
+    migrations_wholesale_repository: MigrationsWholesaleRepository = Mock()
+    migrations_wholesale_repository.read_time_series_points.return_value = time_series_points
+    migrations_wholesale_repository.read_metering_point_periods.return_value = metering_point_periods
+
+    wholesale_internal_repository: WholesaleInternalRepository = Mock()
+    wholesale_internal_repository.read_grid_loss_metering_point_ids.return_value = grid_loss_metering_points
+
+    # Only for wholesale we need these additional tests
+    if "wholesale_calculation" in scenario_path:
```

Comment:
It's ok, but it's more fragile than being more explicit a la this:
```suggestion
    if calculation_args.calculation_type in [CalculationType.WholesaleFixing, CalculationType.WholesaleFirstCorrection, ...):
```